src/sparseml/modifiers/utils/pytorch_helpers.py

```diff
@@ -85,3 +93,41 @@ def run_calibration_forward(
         batch = tensors_to_device(batch, model_device)
         with torch.no_grad():
             forward_fn(batch, module=model)
+        check_for_undercalibrated_modules(
+            batch["input_ids"], model, undercalibrated_module_treshold
+        )
+
+
+def check_for_undercalibrated_modules(
+    batch: torch.Tensor, model: Module, threshold: float
+):
+    """
+    A helper function that warns when a module has seen
+    fewer than threshold % of all the tokens in the batch
+
+    :param batch: the batch of tokens (batch_size, sequence_length)
+    :param model: the model to investigate
+    :param threshold: the minimum percentage of tokens
+        (out of all the tokens in a batch) a module should
+        receive during each forward pass of the calibration
+    """
+    total_token_count = len(batch.flatten())
+    counter = get_observer_token_count(model)
+    for module_name, token_count in counter.items():
+        if token_count is None:
+            # the module has not been observed
+            # or its token_count is not being recorded
+            # by the observer (refer to the observer's
+            # implementation in the source code)
+            continue
+        if token_count / total_token_count < threshold:
+            _LOGGER.warning(
+                f"The module_name: {module_name} "
+                f"received less than {int(threshold * 100)}% "
```

Comment:
let's also add the percent received, or actual `token_count`

Reply:
it's already included in L127